client/runtime/src/main/java/io/quarkiverse/openapi/generator/providers/CredentialsProvider.java

```diff
@@ -12,36 +12,96 @@ public interface CredentialsProvider {
     /**
      * Gets the API Key given the OpenAPI definition and security schema
      *
-     * @param openApiSpecId the OpenAPI Spec identification as defined by the OpenAPI Extension
-     * @param authName The security schema for this API Key definition
+     * @param input the input data available to the method
      * @return the API Key to use when filtering the request
      */
-    String getApiKey(ClientRequestContext requestContext, String openApiSpecId, String authName);
+    String getApiKey(DataInput input);
 
     /**
      * Gets the username given the OpenAPI definition and security schema
      *
-     * @param openApiSpecId the OpenAPI Spec identification as defined by the OpenAPI Extension
-     * @param authName The security schema for this Basic Authorization definition
+     * @param input the input data available to the method
      * @return the username to use when filtering the request
      */
-    String getBasicUsername(ClientRequestContext requestContext, String openApiSpecId, String authName);
+    String getBasicUsername(DataInput input);
 
     /**
      * Gets the password given the OpenAPI definition and security schema
      *
-     * @param openApiSpecId the OpenAPI Spec identification as defined by the OpenAPI Extension
-     * @param authName The security schema for this Basic Authorization definition
+     * @param input the input data available to the method
      * @return the password to use when filtering the request
      */
-    String getBasicPassword(ClientRequestContext requestContext, String openApiSpecId, String authName);
+    String getBasicPassword(DataInput input);
 
     /**
      * Gets the Bearer Token given the OpenAPI definition and security schema
      *
-     * @param openApiSpecId the OpenAPI Spec identification as defined by the OpenAPI Extension
-     * @param authName The security schema for this Bearer Token definition
+     * @param input the input data available to the method
      * @return the Bearer Token to use when filtering the request
      */
-    String getBearerToken(ClientRequestContext requestContext, String openApiSpecId, String authName);
+    String getBearerToken(DataInput input);
+
+    /**
+     * Gets the OAuth2 Bearer Token given the OpenAPI definition and security schema
+     *
+     * @param input the input data available to the method
+     * @return the Bearer Token to use when filtering the request
+     */
+    String getOauth2BearerToken(DataInput input);
+
+    class DataInput {
```

Comment:
```suggestion
    class CredentialsContext {
```

Can you change it to a more specific name? WDYF about `CredentialsContext`?

Reply:
Ah, yes, thx, I struggled with thinking of a good naming here, this one is better than mine!